### PR TITLE
Log error

### PR DESCRIPTION
**Pull request by MalloZup**
# Description

Instruct the user where the installation logs are stored so when it fails he can check the logs

### CODE REVIEW

File: shaptools/netweaver.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def install(
                 cwd=' SAPINST_CWD={}'.format(cwd) if cwd else '')
         result = shell.execute_cmd(cmd, root_user, password, remote_host)
         if result.returncode and raise_exception:
-            raise NetweaverError('SAP Netweaver installation failed')
+            raise NetweaverError('SAP Netweaver installation failed. Logs are saved into installation_folder, default: \"/tmp/swpm_unattended/\" (sapinst_dev.log and sapinst.log)')
```
Comment (Simranpal) [inline shaptools/netweaver.py:246]: The `installation_folder` does not exist in shaptools. May be just write something like: " Please check swpm installation logs (sapinst_dev.log and sapinst.log) located at swpm installation folder for further information. " 
You can mention the default location (  \"/tmp/swpm_unattended/\" )  there maybe

**Review by Simranpal — COMMENTED** [review]
Thanks, Just a comment below

File: shaptools/netweaver.py (hunk 1 above)
Comment (MalloZup) [reply shaptools/netweaver.py:246]: I was actually thinking that we should pass the `installation_folder` as parameter to the function so we can print the exact location. 
So from state we get the pillar and pass to this function.

I think in this case the exact `location` is really important for improving the UX.

Comment (arbulu89) [reply shaptools/netweaver.py:246]: @Simranpal  Why do you say that `/tmp/swpm_unattended` is the default location?
This value is coming from other pieces of cose (salt formula in our case).
@MalloZup I think you are looking for `cwd` parameter. The installation stores the files in that folder (and if if's not set in a predefined folder defined by SAP, that I don't remember now the structure).

In any case, our deployments sets the `cwd` parameter to `/tmp/swpm_unattended` by default

**Review by arbulu89 — CHANGES_REQUESTED** [review]

File: shaptools/netweaver.py (hunk 1 above)
Comment (MalloZup) [reply shaptools/netweaver.py:246]: ok thx @arbulu89  I was thinking also , can we print the `error_msg` of the installer?

something like this in pseudo_code
`Netweaver Install fail. Error: {0}  Check logs at folder{1}".format(error_msg, cwd)`

Comment (Simranpal) [reply shaptools/netweaver.py:246]: @arbulu89 my comment can be improved, I meant to add there"by default"

Comment (MalloZup) [reply shaptools/netweaver.py]: updated thx

File: shaptools/netweaver.py [hunk 2]
```diff
@@ -243,7 +243,7 @@ def install(
                 cwd=' SAPINST_CWD={}'.format(cwd) if cwd else '')
         result = shell.execute_cmd(cmd, root_user, password, remote_host)
         if result.returncode and raise_exception:
-            raise NetweaverError('SAP Netweaver installation failed')
+            raise NetweaverError('SAP Netweaver instal failed. Please check swpm installation logs (sapinst_dev.log and sapinst.log) located at {0} for further information'.format(cwd))
```
Comment (Simranpal) [inline shaptools/netweaver.py:246]: just a typo. `instal/install`

**Review by Simranpal — CHANGES_REQUESTED** [review]
@MalloZup looks good, a comment below and CI failing

**Review by Simranpal — APPROVED** [review]

**Review by arbulu89 — CHANGES_REQUESTED** [review]
And what if `cwd` is `None`? This wouldn't help in any case. If this value is `None`, we would need to prompt other folder. SAP installer creates predefined folder paths for that. I don't remember now the syntax. Maybe @Simranpal knows

**Comment by Simranpal** [comment]
@MalloZup 
By default installer creates `/tmp/sapinst_instdir` directory unless we change it.
We use `setup_cwd` fn to set up alternate directory where installation occurs, and logs are stored. We are setting it to `cwd='/tmp/swpm_unattended'`. In that fn, we are setting default value for `cwd='/tmp/swpm_unattended'`, so may we set default for `cwd` here too. 
What do you think @arbulu89 

I checked the note https://launchpad.support.sap.com/#/notes/0002142461 and 
sap guide section **5.7.1 Useful Information about the Installer** https://help.sap.com/doc/8b7d03f77be6431995d9209c620178e8/CURRENT_VERSION/en-US/inst_SAP_Library_unix.pdf

**Comment by arbulu89** [comment]
> @MalloZup
> By default installer creates `/tmp/sapinst_instdir` directory unless we change it.
> We use `setup_cwd` fn to set up alternate directory where installation occurs, and logs are stored. We are setting it to `cwd='/tmp/swpm_unattended'`. In that fn, we are setting default value for `cwd='/tmp/swpm_unattended'`, so may we set default for `cwd` here too.
> What do you think @arbulu89
> 
> I checked the note https://launchpad.support.sap.com/#/notes/0002142461 and
> sap guide section **5.7.1 Useful Information about the Installer** https://help.sap.com/doc/8b7d03f77be6431995d9209c620178e8/CURRENT_VERSION/en-US/inst_SAP_Library_unix.pdf

@MalloZup So we could use `cwd` if it comes and `/tmp/sapinst_instdir` if it's not. In any case, I would give a check if this is correct, to not just depend on documentation.
We should be able to do so setting the `installation_folder` entry to `None` in the sapnwbootstrap-formula pillar file

**Comment by MalloZup** [comment]
```
            if cwd:
                raise NetweaverError('SAP Netweaver installation failed. Please check swpm installation logs (sapinst_dev.log and sapinst.log) located at {0} for further information'.format(cwd))
            else:
                raise NetweaverError('SAP Netweaver installation failed. Please check swpm installation logs (sapinst_dev.log and sapinst.log) located at /tmp/sapinst_instdir default for further information')

```
something like this

**Comment by arbulu89** [comment]
> ```python
>             if cwd:
>                 raise NetweaverError('SAP Netweaver installation failed. Please check swpm installation logs (sapinst_dev.log and sapinst.log) located at {0} for further information'.format(cwd))
>             else:
>                 raise NetweaverError('SAP Netweaver installation failed. Please check swpm installation logs (sapinst_dev.log and sapinst.log) located at /tmp/sapinst_instdir default for further information')
> ```
> 
> something like this?

Yes, maybe better the 2nd with `.....located at /tmp/sapinst_instdir default folder for further information`.
Please, make sure that none of the lines have more than 100 characters to follow our code quality rules.

File: shaptools/netweaver.py [hunk 3]
```diff
@@ -243,7 +243,12 @@ def install(
                 cwd=' SAPINST_CWD={}'.format(cwd) if cwd else '')
         result = shell.execute_cmd(cmd, root_user, password, remote_host)
         if result.returncode and raise_exception:
-            raise NetweaverError('SAP Netweaver installation failed')
+            if cwd:
+                raise NetweaverError('SAP Netweaver installation failed. Please check swpm installation logs' \
```
Comment (arbulu89) [inline shaptools/netweaver.py:247]: @MalloZup These lines have more than 100 characters. We have the limit in 100. Please, split them more to maximum 100 characters.

**Review by arbulu89 — APPROVED** [review]
@MalloZup Check the characters per line comment.
Ok besides that.

File: shaptools/netweaver.py (hunk 3 above)
Comment (MalloZup) [reply shaptools/netweaver.py:247]: ok will do later on thx!

**Review by arbulu89 — COMMENTED** [review]
@MalloZup About this PR (now merged).
1. I would remove this `print` statement. It's fine for debugging, but more than that it pollutes the test output.
2. The test coverage must have been reduced. The branch without `cwd` is not tested. When you run the tests locally, did you check the test coverage?

**Pull request merged**